modules/hal_rpi_pico/CMakeLists.txt

```diff
@@ -151,6 +151,18 @@ if(CONFIG_HAS_RPI_PICO)
   zephyr_include_directories_ifdef(CONFIG_PICOSDK_USE_CLAIM
           ${common_dir}/hardware_claim/include)
 
+  zephyr_library_sources_ifdef(CONFIG_PICOSDK_USE_RAND
+    ${rp2_common_dir}/hardware_timer/timer.c
+    ${rp2_common_dir}/pico_rand/rand.c
```

Comment:
This code uses pico-sdk's spin-lock feature, not zephyr's one.
I think it can't use as is.
We need to implements zephyr's own TRNG driver.

Reply:
Are you referring to instances such as https://github.com/raspberrypi/pico-sdk/blob/95ea6acad131124694cda1c162c52cd30e0aece0/src/rp2_common/pico_rand/rand.c#L132-L133?

I'm not quite sure what the implications are if using the pico-sdk spinlock. Would it mess with interrupts?

Ideally https://github.com/raspberrypi/pico-sdk/blob/95ea6acad131124694cda1c162c52cd30e0aece0/src/rp2_common/pico_rand/rand.c#L134-L169 is code that I would like to avoid reproducing.